### ISSUE TITLE
Pagination-free UI: one single page with all data sources included

### ISSUE THREAD

**Issue by Dentrax** [medyo/hackertab.dev#70]
Currently, if we follow more than 4 data sources, we need to click _next_ and _prev_ buttons to traverse between pages. (just noticed we could not swipe pages using touchpad with 2-fingers) I thought that the overall UI would be way better if we have an option to toggle _one single page with all data sources included_ feature. One of the caveats is that we might consider reducing the font size to fill all texts in. What do you think? It makes sense? What should we do to get rid of pagination (or at least make it a feature)?

**Comment by medyo** [comment]
I tried to handle the touchpad (2 fingers) but couldn't find an efficient solution, I'll give another try later.
Regarding your second suggestion, do you mean gathering all cards in 1 view (2 rows, 4 columns) or gathering the content of all theses sources in 1 view  ?

**Comment by Dentrax** [comment]
Actually I mean something like: _gathering all cards in 1 view (2 rows, 4 columns)_. But I'm not explicitly saying that way, since I'm not so sure which design will fit better in this case. Here some sample reference, not sure if this adds unnecessary complexity in UI. 🤷‍♂️ What's your thoughts?

![Screen Shot 2021-11-03 at 15 56 40](https://user-images.githubusercontent.com/16493751/140063893-595671e3-a730-40af-b2a3-412e2244617f.png)

**Comment by victor-duarte** [comment]
### Tabs
Tab navigation where each tab is a data source and have normal scroll.
<img width="862" alt="image" src="https://github.com/medyo/hackertab.dev/assets/1746030/a8ad541a-1983-4f45-ae8f-baed80e25717">

### List
Use a list with all the content that can be filtered and sorted
<img width="879" alt="image" src="https://github.com/medyo/hackertab.dev/assets/1746030/ee4af97a-87e9-423e-8cf5-b8b356d58514">

**Comment by medyo** [comment]
Thanks @victor-duarte for the Mockups,

I think going with the `Tabs` layout is totally doable. It's all about making some UI changes, and we can even use it to display the article content later. We just need to add an extra column where users can pick the article they want to see, and when they click on it, it'll show up in the remaining space.

with the 'List' layout, filtering by tags could be challenging. Some sources like Hacker News and Product Hunt don't even use tags, and even when they do, as in the case of Dev.to, the tags aren't always on point.

**Comment by victor-duarte** [comment]
No problem 👍🏻 

@medyo, are you thinking like an email dashboard?

- Folders = data sources
- Emails = data
- Preview = data preview

![image](https://github.com/medyo/hackertab.dev/assets/1746030/1e7e254f-a7f6-4463-8144-5366936c667a)

**Comment by medyo** [comment]
> No problem 👍🏻
> 
> @medyo, are you thinking like an email dashboard?
> 
> * Folders = data sources
> * Emails = data
> * Preview = data preview
> 
> ![image](https://user-images.githubusercontent.com/1746030/275337856-1e7e254f-a7f6-4463-8144-5366936c667a.png)

Yes exactly, the idea is to allow users to read the favourite content without leaving hackertab

**Comment by victor-duarte** [comment]
Will try to create a proof of concept this week and come back with any findings.

**Comment by victor-duarte** [comment]
hey @medyo ,
I was not able to find a way of handling previews due to permissions on source pages, any thoughts?

In the meantime, here is an option/mock on how the layout might look without the preview section

<img width="800" alt="image" src="https://github.com/medyo/hackertab.dev/assets/1746030/d133a5eb-de1b-41f4-b791-b3657216727d">

If we want to proceed with it, we could go into the details on what needs to be changed and updated, e.g:
- would dragging functionality be needed with this new layout
- what data can be shown by row.
- would the compact mode be needed 

etc.

**Comment by medyo** [comment]
@victor-duarte as a beginning this looks nice, is this a coded version or mockup?

**Comment by younes-io** [comment]
@victor-duarte : Looks great! Any updates on this one?

**Comment by victor-duarte** [comment]
@younes-io, no.

Displaying previews (see thread) was not possible due to different domains and I did not got confirmation if this was an issue so I did not pursue.

Will work this week on a PR with the changes of the last screenshot (sidebar + list of sources).

**Comment by victor-duarte** [comment]
@medyo, can you look into the draft PR. There is still work to do in regards to cleaning the cards, but before keep working on it, it would be good to know if its worth keep going down this direction.

Let me know if their is any questions in regards to the implementation details, but in a nutshell is removing the current behavior in large viewports and just re-position elements with css in large viewport.

**Comment by medyo** [comment]
We’ve introduced a new feed display in the latest version, and it should be deployed in the next few days.
